File: OPUS-tauri-app/src-tauri/src/types/workspace.rs
use serde::{Deserialize, Serialize};
use super::error::ErrorEnvelope;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WorkspaceSummary {
    pub name: String,
    pub path: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub created_at: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub last_used: Option<String>,
    #[serde(default)]
    pub archived: bool,
    #[serde(default)]
    pub pinned: bool,
    #[serde(default)]
    pub pin_order: i32,
    #[serde(default)]
    pub expanded: bool,
}

#[derive(Debug, Clone, Serialize)]
pub struct WorkspaceCreateJobRef {
    pub job_id: String,
}

#[derive(Debug, Clone, Serialize)]
pub struct WorkspaceCreateProgress {
    pub job_id: String,
    pub state: String,
    pub repos: Vec<RepoProvisionStatus>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub workspace_name: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub workspace_path: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
pub struct RepoProvisionStatus {
    pub name: String,
    pub state: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub step: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub error: Option<ErrorEnvelope>,
}

File: OPUS-tauri-app/src/types/workspace.ts
import type { ErrorEnvelope } from './errors';

export type WorkspaceSummary = {
  name: string;
  path: string;
  created_at?: string;
  last_used?: string;
  archived: boolean;
  pinned: boolean;
  pin_order: number;
  expanded: boolean;
};

export type WorkspaceCreateJobRef = { job_id: string };

export type WorkspaceCreateProgress = {
  job_id: string;
  state: 'queued' | 'running' | 'succeeded' | 'partial' | 'failed';
  repos: RepoProvisionStatus[];
  workspace_name?: string;
  workspace_path?: string;
};

export type RepoProvisionStatus = {
  name: string;
  state: 'pending' | 'running' | 'succeeded' | 'failed';
  step?: 'preflight' | 'fetch/clone' | 'worktree/checkout' | 'verify';
  error?: ErrorEnvelope;
};
